### ISSUE TITLE
[Bug]: [benchmark][cluster] querynode restart, query and search raise error

### ISSUE THREAD

**Issue by wangting0128** [milvus-io/milvus#15403]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20220126-b9d13962
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.0rc10.dev14
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

client pod: benchmark-no-clean-jhngj-3153285300

[Uploading locust_report_2022-01-26_667.log…]()

client log:
```
[2022-01-27 00:30:23,638] [   ERROR] - RPC error: _execute_search_requests{'RPC start': '2022-01-27 00:29:23.637373', 'RPC error': '2022-01-27 00:30:23.638832'} (pymilvus.decorators:81)
[2022-01-27 00:30:23,639] [   ERROR] - RPC error: search{'RPC start': '2022-01-27 00:29:23.624154', 'RPC error': '2022-01-27 00:30:23.639463'} (pymilvus.decorators:81)
[2022-01-27 00:30:23,640] [   ERROR] - RPC error: describe_collection{'RPC start': '2022-01-27 00:29:23.637662', 'RPC error': '2022-01-27 00:30:23.640509'} (pymilvus.decorators:81)
[2022-01-27 00:31:23,647] [   ERROR] - RPC error: _execute_search_requests{'RPC start': '2022-01-27 00:30:23.646660', 'RPC error': '2022-01-27 00:31:23.647721'} (pymilvus.decorators:81)
[2022-01-27 00:31:23,648] [   ERROR] - RPC error: search{'RPC start': '2022-01-27 00:30:23.640243', 'RPC error': '2022-01-27 00:31:23.648468'} (pymilvus.decorators:81)
[2022-01-27 00:32:23,657] [   ERROR] - RPC error: _execute_search_requests{'RPC start': '2022-01-27 00:31:23.656636', 'RPC error': '2022-01-27 00:32:23.657833'} (pymilvus.decorators:81)
[2022-01-27 00:32:23,658] [   ERROR] - RPC error: search{'RPC start': '2022-01-27 00:31:23.649143', 'RPC error': '2022-01-27 00:32:23.658437'} (pymilvus.decorators:81)
[2022-01-27 00:33:23,665] [   ERROR] - RPC error: _execute_search_requests{'RPC start': '2022-01-27 00:32:23.664653', 'RPC error': '2022-01-27 00:33:23.665886'} (pymilvus.decorators:81)
[2022-01-27 00:33:23,666] [   ERROR] - RPC error: search{'RPC start': '2022-01-27 00:32:23.658943', 'RPC error': '2022-01-27 00:33:23.666457'} (pymilvus.decorators:81)
[2022-01-27 00:33:23,667] [   DEBUG] - Ramping to {"MyUser": 4} (4 total users) (locust.runners:326)
[2022-01-27 00:33:23,667] [   DEBUG] - Spawning additional {"MyUser": 2} ({"MyUser": 2} already running)... (locust.runners:202)
[2022-01-27 00:33:23,667] [   DEBUG] - 4 users spawned (locust.runners:212)
[2022-01-27 00:33:23,667] [   DEBUG] - All users of class MyUser spawned (locust.runners:213)
[2022-01-27 00:33:23,668] [   DEBUG] - 0 users have been stopped, 4 still running (locust.runners:271)
[2022-01-27 00:33:23,669] [    INFO] -  Name                                                                              # reqs      # fails  |     Avg     Min     Max  Median  |   req/s failures/s (locust.stats_logger:725)
[2022-01-27 00:33:23,669] [    INFO] - ---------------------------------------------------------------------------------------------------------------------------------------------------------------- (locust.stats_logger:727)
[2022-01-27 00:33:23,670] [    INFO] -  grpc query                                                                             4   4(100.00%)  |   60009   60007   60015   60007  |    0.00    0.00 (locust.stats_logger:730)
[2022-01-27 00:33:23,670] [    INFO] - ---------------------------------------------------------------------------------------------------------------------------------------------------------------- (locust.stats_logger:731)
[2022-01-27 00:33:23,670] [    INFO] -  Aggregated                                                                             4   4(100.00%)  |   60009   60007   60015   60007  |    0.00    0.00 (locust.stats_logger:732)
[2022-01-27 00:33:23,670] [    INFO] -  (locust.stats_logger:733)
[2022-01-27 00:34:23,676] [   ERROR] - RPC error: _execute_search_requests{'RPC start': '2022-01-27 00:33:23.675251', 'RPC error': '2022-01-27 00:34:23.676377'} (pymilvus.decorators:81)
[2022-01-27 00:34:23,677] [   ERROR] - RPC error: search{'RPC start': '2022-01-27 00:29:23.625418', 'RPC error': '2022-01-27 00:34:23.677055'} (pymilvus.decorators:81)
[2022-01-27 00:39:23,677] [   ERROR] - RPC error: query{'RPC start': '2022-01-27 00:33:23.671471', 'RPC error': '2022-01-27 00:39:23.676977'} (pymilvus.decorators:81)
[2022-01-27 00:39:23,678] [   ERROR] - RPC error: describe_collection{'RPC start': '2022-01-27 00:34:23.676200', 'RPC error': '2022-01-27 00:39:23.678359'} (pymilvus.decorators:81)
[2022-01-27 00:39:23,678] [   ERROR] - RPC error: describe_collection{'RPC start': '2022-01-27 00:34:23.676313', 'RPC error': '2022-01-27 00:39:23.678849'} (pymilvus.decorators:81)
[2022-01-27 00:44:23,683] [   ERROR] - RPC error: query{'RPC start': '2022-01-27 00:39:23.676771', 'RPC error': '2022-01-27 00:44:23.682978'} (pymilvus.decorators:81)
```

### Expected Behavior

argo task: benchmark-no-clean-jhngj

test yaml:
client-configmap:client-random-locust-search-1b-240h
server-configmap:server-cluster-8c64m-datanode2-indexnode4-querynode6-pulsar2.8.2

server:
```
NAME                                                              READY   STATUS      RESTARTS   AGE    IP             NODE                      NOMINATED NODE   READINESS GATES
benchmark-no-clean-jhngj-1-etcd-0                                 1/1     Running     0          17h    10.97.16.209   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-etcd-1                                 1/1     Running     0          17h    10.97.17.251   qa-node014.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-etcd-2                                 1/1     Running     0          17h    10.97.16.211   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-datacoord-75ccc67ddf-tlbwc      1/1     Running     0          17h    10.97.14.74    qa-node011.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-datanode-6d7f97b9c8-tttss       1/1     Running     0          17h    10.97.11.204   qa-node009.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-datanode-6d7f97b9c8-x77rp       1/1     Running     0          17h    10.97.17.248   qa-node014.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-indexcoord-99bbbdbfb-kqqnv      1/1     Running     0          17h    10.97.19.152   qa-node016.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-indexnode-c7bb97fdc-55xf5       1/1     Running     0          17h    10.97.19.153   qa-node016.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-indexnode-c7bb97fdc-74mst       1/1     Running     0          17h    10.97.14.75    qa-node011.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-indexnode-c7bb97fdc-mgtv5       1/1     Running     0          17h    10.97.14.76    qa-node011.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-indexnode-c7bb97fdc-xvrfl       1/1     Running     0          17h    10.97.10.227   qa-node008.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-proxy-ddf788dbd-z8gh7           1/1     Running     0          17h    10.97.3.112    qa-node001.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-querycoord-c6d4f7d75-jk5fn      1/1     Running     0          17h    10.97.14.78    qa-node011.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-querynode-56fd46f97c-7kvzb      1/1     Running     0          17h    10.97.20.2     qa-node018.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-querynode-56fd46f97c-d24j7      1/1     Running     0          17h    10.97.16.202   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-querynode-56fd46f97c-j5cfx      1/1     Running     0          17h    10.97.17.249   qa-node014.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-querynode-56fd46f97c-k7jrk      1/1     Running     1          17h    10.97.19.154   qa-node016.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-querynode-56fd46f97c-n6zn7      1/1     Running     0          17h    10.97.16.201   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-querynode-56fd46f97c-qhvtk      1/1     Running     0          17h    10.97.20.3     qa-node018.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-milvus-rootcoord-7d5b88ff49-vpvrm      1/1     Running     0          17h    10.97.3.111    qa-node001.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-minio-0                                1/1     Running     0          17h    10.97.20.5     qa-node018.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-minio-1                                1/1     Running     0          17h    10.97.16.213   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-minio-2                                1/1     Running     0          17h    10.97.16.208   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-minio-3                                1/1     Running     0          17h    10.97.16.212   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-autorecovery-7d7967f9f6-mmgkb   1/1     Running     0          17h    10.97.16.198   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-bastion-7856fb999d-4c4wr        1/1     Running     0          17h    10.97.16.199   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-bookkeeper-0                    1/1     Running     0          17h    10.97.16.206   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-bookkeeper-1                    1/1     Running     0          17h    10.97.16.214   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-bookkeeper-2                    1/1     Running     0          17h    10.97.16.215   qa-node013.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-broker-5b4f484b7d-xj2w5         1/1     Running     0          17h    10.97.3.113    qa-node001.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-proxy-6569f55fd4-bn9bf          2/2     Running     0          17h    10.97.4.47     qa-node002.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-zookeeper-0                     1/1     Running     0          17h    10.97.4.48     qa-node002.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-zookeeper-1                     1/1     Running     0          17h    10.97.4.49     qa-node002.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-zookeeper-2                     1/1     Running     0          17h    10.97.4.50     qa-node002.zilliz.local   <none>           <none>
benchmark-no-clean-jhngj-1-pulsar-zookeeper-metadata-dkcqp        0/1     Completed   0          17h    10.97.16.200   qa-node013.zilliz.local   <none>           <none>
```

### Steps To Reproduce

```markdown
1、create collection
2、build index of ivf_sq8
3、insert 1 billion vectors
4、flush collection
5、build index with the same params
6、load collection
7、locust concurrent: query<-search, load, get<-query 《- raise error
```

### Anything else?

client-random-locust-search-1b-240h:
```
locust_random_performance:
      collections:
        -
          collection_name: sift_1b_128_l2
          ni_per: 50000
          build_index: true
          index_type: ivf_sq8
          index_param:
            nlist: 1024
          task:
            types:
              -
                type: query
                weight: 20
                params:
                  top_k: 10
                  nq: 10
                  search_param:
                    nprobe: 16
              -
                type: load
                weight: 1
              -
                type: get
                weight: 2
                params:
                  ids_length: 10
            connection_num: 1
            clients_num: 20
            spawn_rate: 2
            during_time: 240h
```

**Comment by xige-16** [comment]
/unassign
@XuanYang-cn is  looking into the issue.

**Comment by XuanYang-cn** [comment]
```
24333 [2022/01/27 00:29:23.864 +00:00] [DEBUG] [segment.go:311] ["do search on segment"] [segmentID=430761690400292865] [segmentType=3]
24334 [2022/01/27 00:29:23.865 +00:00] [DEBUG] [segment.go:311] ["do search on segment"] [segmentID=430754970260209665] [segmentType=3]
24335 fatal error: unexpected signal during runtime execution                                                                                      
24336 runtime stack:                                                                                                                               
24337 runtime.sigpanic()                                                                                                                           
24338 goroutine 200863 [syscall]:                                                                                                                  
24339     /usr/local/go/src/runtime/signal_unix.go:704 +0x4ac                                                                                      
24340                                                                                                                                              
24341 [signal SIGSEGV: segmentation violation code=0x80 addr=0x0 pc=0x7f2d4fbdf683]                                                                
24342 runtime.throw(0x33238e6, 0x2a)                                                                                                               
24343                                                                                                                                              
24344     /usr/local/go/src/runtime/panic.go:1116 +0x72                                                                                            
24345 github.com/milvus-io/milvus/internal/querynode._Cfunc_Search(0x5d4b6020, 0x7f2a7fffaae0, 0x7f278c000030, 0x5fa63bf74580005, 0xc00312d548, 0x5fa5fd4bc400001, 0xc000000000, 0x0)
24346     /go/src/github.com/milvus-io/milvus/internal/querynode/segment.go:312 +0x125 fp=0xc000596da8 sp=0xc000596d40 pc=0x1bbef05
24347     /go/src/github.com/milvus-io/milvus/internal/querynode/segment.go:312 +0x3c5 fp=0xc000596ee0 sp=0xc000596da8 pc=0x1bb4985
24348     /usr/local/go/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc000596f88 sp=0xc000596f80 pc=0x835d01               
24349                                                                                                                                              
24350     /go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:495 +0x445                                                                        
24351     /go/src/github.com/milvus-io/milvus/cmd/main.go:296 +0x635                                                                               
24352                                                                                                                                              
24353 github.com/apache/pulsar-client-go/pulsar.(*negativeAcksTracker).track(0xc0008fb620)                          
24354     /go/pkg/mod/github.com/milvus-io/pulsar-client-go@v0.6.5/pulsar/negative_acks_tracker.go:81 +0x5aa        
24355 created by github.com/apache/pulsar-client-go/pulsar.newNegativeAcksTracker 
```

**Comment by xiaofan-luan** [comment]
> ```
> 24333 [2022/01/27 00:29:23.864 +00:00] [DEBUG] [segment.go:311] ["do search on segment"] [segmentID=430761690400292865] [segmentType=3]
> 24334 [2022/01/27 00:29:23.865 +00:00] [DEBUG] [segment.go:311] ["do search on segment"] [segmentID=430754970260209665] [segmentType=3]
> 24335 fatal error: unexpected signal during runtime execution                                                                                      
> 24336 runtime stack:                                                                                                                               
> 24337 runtime.sigpanic()                                                                                                                           
> 24338 goroutine 200863 [syscall]:                                                                                                                  
> 24339     /usr/local/go/src/runtime/signal_unix.go:704 +0x4ac                                                                                      
> 24340                                                                                                                                              
> 24341 [signal SIGSEGV: segmentation violation code=0x80 addr=0x0 pc=0x7f2d4fbdf683]                                                                
> 24342 runtime.throw(0x33238e6, 0x2a)                                                                                                               
> 24343                                                                                                                                              
> 24344     /usr/local/go/src/runtime/panic.go:1116 +0x72                                                                                            
> 24345 github.com/milvus-io/milvus/internal/querynode._Cfunc_Search(0x5d4b6020, 0x7f2a7fffaae0, 0x7f278c000030, 0x5fa63bf74580005, 0xc00312d548, 0x5fa5fd4bc400001, 0xc000000000, 0x0)
> 24346     /go/src/github.com/milvus-io/milvus/internal/querynode/segment.go:312 +0x125 fp=0xc000596da8 sp=0xc000596d40 pc=0x1bbef05
> 24347     /go/src/github.com/milvus-io/milvus/internal/querynode/segment.go:312 +0x3c5 fp=0xc000596ee0 sp=0xc000596da8 pc=0x1bb4985
> 24348     /usr/local/go/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc000596f88 sp=0xc000596f80 pc=0x835d01               
> 24349                                                                                                                                              
> 24350     /go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:495 +0x445                                                                        
> 24351     /go/src/github.com/milvus-io/milvus/cmd/main.go:296 +0x635                                                                               
> 24352                                                                                                                                              
> 24353 github.com/apache/pulsar-client-go/pulsar.(*negativeAcksTracker).track(0xc0008fb620)                          
> 24354     /go/pkg/mod/github.com/milvus-io/pulsar-client-go@v0.6.5/pulsar/negative_acks_tracker.go:81 +0x5aa        
> 24355 created by github.com/apache/pulsar-client-go/pulsar.newNegativeAcksTracker 
> ```

do we have core dump right now for this issue?

**Comment by XuanYang-cn** [comment]
No core dump in benchmark

- [ ] feature: Benchmark needs to generate core dump and store it once service panicked

**Comment by XuanYang-cn** [comment]
Possibly the same as #15133 , #14077 , but this time `search segmentation violation` happened on cluster querynode.

**Comment by yanliang567** [comment]
remove urgent label, as it #15133 was fixed in 2.0.1
